### ISSUE TITLE
Enable download of OSM notes for offline use

### ISSUE THREAD

**Issue by daganzdaanda** [osmandapp/Osmand#4424]
(somewhat related to #2445)

Hi! 
It would be nice to be able to download OSM notes, and keep them available when there is no internet connection, so I can go check out the issue locally any time without data connection. 

Maybe it would be enough if the OSM notes could be cached for a certain time, maybe 24h? Or just keep the notes offline until the next time the device is online. (Maybe after a day, the notes icon could turn grey to show that it might be out of sync.)

Alternatively, there could be an entry for "OSM notes" in the same menu as the maps downloads. The complete "planet-notes" dump for the whole world is currently a bit more that 50 MB.

In any case, if the user replys / closes a note while offline, a refresh would be necessary before uploading the changes.

BTW, thanks a lot for OsmAnd, everyone! I'm especially happy that it still works and gets updated on my old device with android 4.1.2 :-)

**Comment by aceman444** [comment]
This would be great.

**Comment by int-ua** [comment]
Can it be implemented just by not hiding all notes that are already shown in case request for new ones showed no connection?

**Comment by InsertUser** [comment]
There have been a few times when I would have been able to review several notes if they had remained on the map when I left WiFi range. 

Please implement this if you can.

**Comment by vshcherb** [comment]
Yes we should make a map out of https://planet.openstreetmap.org/notes/planet-notes-latest.osn.bz2 though it probably will be updated once a month in the beginning.

**Comment by NotSoImportant** [comment]
I like to contribute more to osm. Using OsmAnd to do so with offline maps. Going arround and mapping. However when I return back to the PC I see there had been some notes next to me I could have checked or resolved. Unfortunately without internet connection I was not able to see them.

Proposed solution: 
When internet connection is lost - just keep the already known notes visible. On internet reconnection this could be refreshed.
(simiar to https://github.com/osmandapp/Osmand/issues/4424#issuecomment-348919713)

Getting notes on a monthly source will not be helpful. Data should be real-timeish (e.g. updated once a day or whener there is a internet connection). Also I don't need ALL notes, but only of the maps I have currently activated.

**Comment by MiMoHo** [comment]
Still very much wanted: when the connection drops, notes silently disappear from the map, which makes them look resolved/removed while surveying — exactly the moment you need them.

A concrete implementation proposal to make this actionable:

1. **Download scope coupled to installed maps**: when (re)loading the OSM notes layer online, fetch notes via the Notes API for the bounding boxes of the downloaded/installed maps (optionally: only regions with "Live updates" enabled), store them locally (SQLite next to the existing local notes DB).
2. **Offline fallback**: when the layer has no connectivity, render the cached notes (with an "as of <date>" hint) instead of nothing.
3. **Refresh policy**: refresh a region's cache when the notes layer is used online, plus a manual "update notes" action; cached entries older than a configurable TTL get greyed out rather than hidden.

That keeps traffic bounded (no world-wide sync), matches user expectation ("I have this map, so I see its notes"), and reuses the existing notes rendering. I'd be happy to work on a PR along these lines if the approach is acceptable.
